lucene/core/src/java/org/apache/lucene/search/Weight.java

```diff
@@ -204,9 +204,14 @@ public int score(LeafCollector collector, Bits acceptDocs, int min, int max) thr
       collector.setScorer(scorer);
       DocIdSetIterator scorerIterator = twoPhase == null ? iterator : twoPhase.approximation();
       DocIdSetIterator collectorIterator = collector.competitiveIterator();
-      // if possible filter scorerIterator to keep only competitive docs as defined by collector
-      DocIdSetIterator filteredIterator = collectorIterator == null ? scorerIterator :
-          ConjunctionDISI.intersectIterators(Arrays.asList(scorerIterator, collectorIterator));
+      DocIdSetIterator filteredIterator = scorerIterator;
+      if (collectorIterator != null) {
+        if (scorerIterator.docID() != -1) {
+          collectorIterator.advance(scorerIterator.docID());
+        }
```

Comment:
I don't think that this is good enough as we might be advancing ahead of scorerIterator? This was why I thought that we should instead wrap scorerIterator in such a way that its initial docID would be -1.

Reply:
@jpountz  Addressed in cba6cf75f48a26cd48f09152c25518c91fd6660d

Reply:
oh, I had not considered setting `scorerIterator.docID()` as a min docID, maybe this means that we no longer need the `min` parameter of `RangeDISIWrapper`?

Reply:
Thanks @jpountz , addressed in d42c4649c81364f13c51c0b147dd600e57d7cccc